logstash-core/src/main/java/org/logstash/instrument/witness/process/ProcessWitness.java

```diff
@@ -0,0 +1,247 @@
+            gen.writeObjectFieldStart(KEY);
+            numberSerializer.serialize(witness.openFileDescriptors);
+            numberSerializer.serialize(witness.peakOpenFileDescriptors);
+            numberSerializer.serialize(witness.maxFileDescriptors);
+            //memory
+            gen.writeObjectFieldStart(Memory.KEY);
+            numberSerializer.serialize(witness.memory.memTotalVirtualInBytes);
+            gen.writeEndObject();
+            //cpu
+            gen.writeObjectFieldStart(Cpu.KEY);
+            numberSerializer.serialize(witness.cpu.cpuTotalInMillis);
+            numberSerializer.serialize(witness.cpu.cpuProcessPercent);
+
+            //TODO: jake load average
+
+            gen.writeEndObject();
+            gen.writeEndObject();
+        }
+    }
+
+    /**
+     * The Process snitch. Provides a means to get discrete metric values.
+     */
+    public static class Snitch {
+
+        private final ProcessWitness witness;
+
+        private Snitch(ProcessWitness witness) {
+            this.witness = witness;
+        }
+
+        /**
+         * Get the number of open file descriptors for this process
+         *
+         * @return the open file descriptors
+         */
+        public long openFileDescriptors() {
+            return witness.openFileDescriptors.getValue().longValue();
+        }
+
+        /**
+         * Get the max file descriptors for this process
+         *
+         * @return the max file descriptors
+         */
+        public long maxFileDescriptors() {
+            return witness.maxFileDescriptors.getValue().longValue();
+        }
+
+        /**
+         * Get the high water number of open file descriptors for this process
+         *
+         * @return the high water/ peak of the seen open file descriptors
+         */
+        public long peakOpenFileDescriptors() {
+            return witness.peakOpenFileDescriptors.getValue().longValue();
+        }
+
+        /**
+         * Get the cpu percent for this process
+         *
+         * @return the cpu percent
+         */
+        public short cpuProcessPercent() {
+            return witness.cpu.cpuProcessPercent.getValue().shortValue();
+        }
+
+        /**
+         * Get the total time of the cpu in milliseconds for this process
+         *
+         * @return the cpu total in milliseconds
+         */
+        public long cpuTotalInMillis() {
+            return witness.cpu.cpuTotalInMillis.getValue().longValue();
+        }
+
+        /**
+         * Get the committed (virtual) memory for this process
+         *
+         * @return the committed memory
+         */
+        public long memTotalVirtualInBytes() {
+            return witness.memory.memTotalVirtualInBytes.getValue().longValue();
+        }
+
+        /**
+         * Gets if this process is running on *nix based system.
+         *
+         * @return true if host is *nix, false otherwise
+         */
+        public boolean isUnix() {
```

Comment:
@jakelandis not sure we should have this method at all, every use of it could just be replaced by a `static` constant couldn't it? (just trying to save some code size:))

Reply:
will do